app/lib/cluster_viz_service.rb

```diff
@@ -250,21 +220,24 @@ def self.load_cluster_group_data_array_points(study, cluster, annotation, subsam
 
   # returns an array of the values for the given annotation, sorted by the cells of the given cells array
   def self.get_annotation_values_array(study, cluster, annotation, cells, subsample_annotation, subsample_threshold)
-    annotation_array = []
-    annotation_hash = {}
+    return [] if study.nil? || cluster.nil?
+
     # Construct the arrays based on scope
-    if annotation[:scope] == 'cluster'
+    case annotation[:scope]
```

Comment:
if/elsif can be clearer than `case` in situations like this where the number of  alternatives is small, because it can be challenging to remember how case/when behaves with respect to needing explicit break statements, since `case` behaves differently in different languages.

Reply:
Agreed, but rubocop was barking at me uniformly about converting all `if/elsif/else` blocks into case statements.